### ISSUE TITLE
Why does negative optimization occur on my dataset

### ISSUE THREAD

**Issue by Tartisan** [hku-mars/HBA#38]
![7fd41626-2217-4a31-9459-5452ee3b2647](https://github.com/user-attachments/assets/f36bff52-647d-46f2-8048-e536153d39c6)
above is the initial effect

![img_v3_02el_cb6ac999-c915-4a99-a8b8-e2b00375187g](https://github.com/user-attachments/assets/b54f1f8f-e646-4962-9f8f-b8ba2a9dd771)
this is the effect after HBA
I don't know why. I also tried BALM, it's ok for this case, but the memory consumption is too high. 
Looking forward for reply, thx!

**Comment by chengwei0427** [comment]
Hi， @Tartisan 
There might be a bug in the write_pose function. I've submitted a [PR](https://github.com/hku-mars/HBA/pull/39/commits/9e2960fb2abd0f72ef9fa1638e5a521c4ec51a53) for it, and you can take a look at it for reference.

**Comment by G4419** [comment]
@chengwei0427 Yes, you are correct. The original write_pose function set first pose to Identity pose, which is wrong.
